src/resources/views/crud/filters/select2_ajax.blade.php

```diff
@@ -89,21 +104,32 @@ class="nav-item dropdown {{ Request::get($filter->name)?'active':'' }}">
 				        url: '{{ $filter->values }}',
 				        dataType: 'json',
 				        type: 'GET',
-				        quietMillis: 50,
-				        // data: function (term) {
-				        //     return {
-				        //         term: term
-				        //     };
-				        // },
+				        quietMillis: $(this).attr('filter-quiet-time'),
+
 				        processResults: function (data) {
-				            return {
-				                results: $.map(data, function (item, i) {
-				                    return {
-				                        text: item,
-				                        id: i
-				                    }
-				                })
-				            };
+                            //it's a paginated result
+                            if(Array.isArray(data.data)) {
+                                if(data.data.length > 0) {
+                               return {
+                                    results: $.map(data.data, function (item) {
+                                     return {
+                                        text: item[selectAttribute],
+                                        id: item[selectKey]
+                                    }
+                                })
+                                };
+                                }
+                            }else{
+                                //it's non-paginated result
+                                return {
+                                    results: $.map(data, function (item, i) {
+                                        return {
+                                            text: item,
+                                            id: i
+                                        }
+                                    })
+                                };
+                            }
```

Comment:
```suggestion
                            if(Array.isArray(data.data)) {
                                // it's a paginated result
                                if(data.data.length > 0) {
                                    return {
                                        results: data.data.map(item => ({
                                            id: item[selectKey],
                                            text: item[selectAttribute],
                                        }))
                                    };
                                }
                            } else {
                                // it's non-paginated result
                                return {
                                    results: data.map((text, id) => ({
                                        id,
                                        text,
                                    }))
                                };
                            }
```
Formated, and removed dependency of jquery map since vanilla map acts the same way.